src/main/haskell/kore/src/Kore/Step/Step.hs

```diff
@@ -123,7 +126,9 @@ transitionRule tools substitutionSimplifier simplifier =
             $ stepWithRewriteRule tools substitutionSimplifier config a
         case result of
             Left _ -> pure []
-            Right results -> return $ mapMaybe (patternFromResult proof) results
+            Right results ->
+                Log.withLogScope "transitionRule" $
```

Comment:
Can we easily get a source location here? There are already multiple things called `transitionRule`.